components/nexus-repository/src/test/java/org/sonatype/nexus/repository/negativecache/NegativeCacheFacetImplTest.groovy

```diff
@@ -0,0 +1,278 @@
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.negativecache
+
+import com.google.common.collect.Lists
+import com.google.common.collect.Maps
+import net.sf.ehcache.Cache
+import net.sf.ehcache.CacheManager
+import net.sf.ehcache.Ehcache
+import net.sf.ehcache.Element
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentCaptor
+import org.sonatype.nexus.repository.Repository
+import org.sonatype.nexus.repository.config.Configuration
+import org.sonatype.nexus.repository.http.HttpStatus
+import org.sonatype.nexus.repository.util.NestedAttributesMap
+import org.sonatype.nexus.repository.view.Status
+import org.sonatype.sisu.goodies.common.Time
+import org.sonatype.sisu.goodies.eventbus.EventBus
+import org.sonatype.sisu.litmus.testsupport.TestSupport
+
+import static net.sf.ehcache.Status.STATUS_ALIVE
+import static net.sf.ehcache.Status.STATUS_SHUTDOWN
+import static org.mockito.Matchers.any
+import static org.mockito.Mockito.mock
+import static org.mockito.Mockito.never
+import static org.mockito.Mockito.verify
+import static org.mockito.Mockito.when
+
+/**
+ * Tests for {@link NegativeCacheHandler}.
+ */
+class NegativeCacheFacetImplTest
+extends TestSupport
+{
+  private NegativeCacheFacetImpl underTest
+  private NegativeCacheKey key
+  private Status status
+  private CacheManager cacheManager
+  private Ehcache cache
+  private Repository repository
+  private Map attributes
+
+  @Before
+  void setUp() {
+    cacheManager = mock(CacheManager)
+    when(cacheManager.status).thenReturn(STATUS_ALIVE)
+    cache = mock(Ehcache)
+    underTest = new NegativeCacheFacetImpl(cacheManager) {
+      @Override
+      Ehcache createCache(final String name, final Time timeToLive) {
+        assert name == 'negative-cache-test'
+        assert timeToLive.equals(Time.minutes(1440))
+        when(cache.name).thenReturn(name)
+        return cache
+      }
+    }
+    underTest.installDependencies(mock(EventBus))
+    key = mock(NegativeCacheKey)
+    status = Status.failure(HttpStatus.NOT_FOUND, '404')
+    repository = mock(Repository)
+    when(repository.name).thenReturn('test')
+    Configuration config = mock(Configuration)
+    when(repository.configuration).thenReturn(config)
+    when(config.attributes(NegativeCacheFacetImpl.CONFIG_KEY)).thenReturn(
+        new NestedAttributesMap(NegativeCacheFacetImpl.CONFIG_KEY, attributes = Maps.newHashMap())
+    )
+  }
+
+  /**
```

Comment:
It's a shame that Spock and Mockito don't play very nicely together, as the given/then style is very nice in Spock.